.github/CONTRIBUTING.md

```diff
@@ -1,4 +1,47 @@
-# How to Contribute
+# Contributing
+
+This library's source code is mainly generated from JSON schema in the [Google CloudEvents repo](https://github.com/googleapis/google-cloudevents).
+
+Specifically, the `google-cloudevents-types/src` directory is generated and should not be modified as changes will be overwritten when the library is re-generated. Rather, the upstream CloudEvents repo should be modified.
+
+## Generate API types
+
+Before generating the library, you must set up your environment like such:
+- Have Java 11+ and Node (for generation) installed.
+- Install the [quicktype wrapper CLI](https://github.com/googleapis/google-cloudevents/tree/master/tools/quicktype-wrapper) using npm.
+- Clone the [google-cloudevents](https://github.com/googleapis/google-cloudevents) source of truth repo in the same directory as this repo.
+
+Then, to generate, run the ./gen.sh file in the java repo. This will generate the whole library with the source of truth JSON schemas.
+
+```sh
+./gen.sh
+```
+
+Observe the new files in `google-cloudevent-types/src/`.
+
+## Test
+
+Run tests in the `google-cloudevent-types` :
+
+```sh
+cd google-cloudevent-types
+mvn install
+mvn test
+```
+
+## Generate a New Artifact
+
+A new artifact is unlikely. Here is a generic script to generate a new artifact:
+
+```sh
+mvn -B archetype:generate \
```

Comment:
This creates a new maven project. Is this needed when running the generator?

Reply:
It's unlikely we need a new maven project, but it's here in case.

Running the generator is described above.

Reply:
That doesn't generate a *new* artifact though, it generates another one called google-cloudevent-types. I suspect it would be clearer just to remove this section of the document. Anyone who really has needs to create a new, separate artifact is likely to know more information than this anyway.

Reply:
Removed.